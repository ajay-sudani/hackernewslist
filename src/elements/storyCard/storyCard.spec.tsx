import React from "react";
import renderer from "react-test-renderer";
import StoryCard, { IStroyProps } from "./storyCard";

const storyCard: IStroyProps = {
  by: "John",
  score: 85,
  time: 1234567890,
  title: "Latest news about W3C",
  type: "Blog",
  url: "http://www.dummy.com",
};

describe("StoryCard", () => {
  it("should render with default props", () => {
    const animateElement = renderer
      .create(<StoryCard {...storyCard} />)
      .toJSON();
    expect(animateElement).toMatchSnapshot();
  });
});
