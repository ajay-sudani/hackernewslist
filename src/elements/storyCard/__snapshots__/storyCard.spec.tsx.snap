// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryCard should render with default props 1`] = `
<div>
  <div
    style={
      Object {
        "opacity": 0,
        "transform": "translateY(2rem) translateZ(0)",
      }
    }
  >
    <div
      aria-label="news"
      role="group"
    >
      <a
        href="http://www.dummy.com"
        rel="noreferrer"
        target="_blank"
      >
        <p>
          Blog
        </p>
        <p>
          John
        </p>
        <p>
          Score: 
          85
        </p>
        <p>
          Sat Feb 14 2009
        </p>
        <p>
          Latest news about W3C
        </p>
      </a>
    </div>
  </div>
</div>
`;
